Added node and yarn dev setup

matthewcn56 opened the pull request:
Our projects like the Editor and the Learning Labs will need either npm or yarn for package management, so this adds in the info needed to install them inside of `dev-setup.md`

mattxwang commented:
My two cents as a retiree - might be better to make this two separate pages, i.e.

* setting up your environment for the website
* setting up your environment for editor/learning labs 

Ngl the docs are a bit of a mess, so feel free to revamp them!

matthewcn56 commented:
> My two cents as a retiree - might be better to make this two separate pages

Sounds good! With a bit of a reorganization now...
- `dev-setup.md` is the base page for set-up, and links to and from the `project-setup` page as well.
-  Changed _Add To Team Page_ to _Website Setup_ and added in _Project Setup_ to both the footer and sidebar.
- Changed the `index.html` of _docs to support both website and project setup.

mattxwang reviewed (changes requested):
I'll leave a more thorough review after I head off to Seattle (if you'd like), but my quick note is that I think the current setup is still a bit confusing: all the information about Jekyll, WSL, bundle, etc. doesn't need to be in "dev setup" since it's only related to Jekyll and the Ruby stack.

If I could redo the structure (wink wink, nudge nudge) I'm thinking:

* developer setup -> just how to use git, etc.
* website setup -> ruby, bundle, git review, wsl stuff
* website first-steps -> this is the adding yourself to team page
* node setup -> what you have now, but under "node" may make more sense. also, you may want to explain how to install node and/or nvm

matthewcn56 commented:
Thanks for the feedback @mattxwang! I think that we can take out `dev-setup.md` and just discuss the Git stuff inside of _doc's `index.html` and making that be the default dev-setup page, and separating all the ruby, Jekyll etc stuff onto the website setup page would help differentiate between project vs. website set-up better! I def agree that rn it's a bit confusing of where everything is

matthewcn56 commented:
Restructured the pages more by combining static website set-up and first steps, moving all the git stuff into `dev-setup`, changed what `Dev Setup` in the footer linked to, updated pages to include the Dev Resources Page

mattxwang commented on _docs/dev-setup.md:
```diff
@@ -91,88 +89,59 @@ For OSX, the installation process will seem rather complicated, but you should e
 We want to make sure that changes we make don't impact other people, or at least not until those changes are done. Out of courtesy, we'll develop our changes on our own "branch", which is a version of our code.
 
-The default branch is called `master`, and it contains the code that goes live on [teachla.uclaacm.com](https://teachla.uclaacm.com). We don't want to touch that just yet, so we're going to make our own branch!
+Default branches are either called `master` or `main`, and they contain the code that goes live on our websites' links! For instance, `teach-la-website`'s master branch code can be seen on [teachla.uclaacm.com](https://teachla.uclaacm.com). We don't want to touch that just yet, so we're going to make our own branch!
```
In the future, we should shift all of our branches to `main` - makes writing this a lot easier.

mattxwang commented on _docs/project-setup.md:
````diff
@@ -0,0 +1,102 @@
+```
+
+If you already have node installed in your machine, you should see some
+numbers in the form of `vXX.XX.X` which means that you already have node (and consequently npm) installed, and don't need to do anything else!
````
Might want to add something about `npm -v` too!

mattxwang replied:
Also, we should probably mention something about using the right node version - esp with node 16, it really does matter which version they install.

mattxwang commented on _docs/project-setup.md:
````diff
@@ -0,0 +1,102 @@
+
+If you want to use [node version manager](https://github.com/nvm-sh/nvm) instead, be sure to checkout our guide to it in the appendix!
+
+To check that node has been properly installed on your machine, just run:
+
+```sh
+$ node -v
+```
+
````
Is this duped?

mattxwang commented on _docs/static-setup.md:
````diff
@@ -1,26 +1,119 @@
 ```sh
-$ git checkout -b my-branch-name    # create a new branch and check it out
-$ jekyll serve                      # set up our dev server
+$ git clone https://github.com/uclaacm/teach-la-website.git
````
Ooh, I'd use the ssh clone by default if possible. Might be good to recommend the ssh keygen stuff as well in the dev setup.

mattxwang commented on _docs/static-setup.md:
````diff
@@ -1,26 +1,119 @@
+
+```sh
+$ git checkout -b my-branch-name    # create a new branch and check it out
+$ jekyll serve                      # set up our dev server
````
this should be `bundle exec jekyll serve`

mattxwang reviewed (changes requested):
Think this is looking good. Most of these comments are nitpicky, and I would also suggest requesting review from someone on the teach la dev team!

Also, a generic comment that has too many spotchecks - I would strongly avoid using "here" as link text, since it's not very accessible. Try rewording the language so it's more clear what the link is going towards!

reginawang99 commented:
@matthewcn56 any way i can help?

matthewcn56 commented:
So sorry about that, will finish this up this weekend! Just have to move around a couple files

reginawang99 commented:
Oh no worries! didnt know if you were waiting on vivian and i could help or something :))
